src/map/script.c

```diff
@@ -8616,6 +8616,70 @@ static BUILDIN(delitem2)
 	return false;
 }
 
+/**
+ * Deletes item at given index.
+ * delitem(<index>{, <account id>});
+ */
+static BUILDIN(delitemidx)
+{
+	struct map_session_data *sd;
+
+	if (script_hasdata(st, 4)) {
+		if ((sd = script->id2sd(st, script_getnum(st, 4))) == NULL) {
+			st->state = END;
+			return true;
+		}
+	} else {
+		if ((sd = script->rid2sd(st)) == NULL)
+			return true;
+	}
+
+	int i = script_getnum(st, 2);
+	if (i < 0 || i >= sd->status.inventorySize) {
+		ShowError("buildin_delitemidx: Index (%d) should be from 0-%d.\n", i, sd->status.inventorySize - 1);
+		st->state = END;
+		return false;
+	}
+
+	struct item it;
+	memset(&it, 0, sizeof(it));
+	it.nameid = sd->status.inventory[i].nameid;
+	struct item_data *item_data = itemdb->exists(it.nameid);
+	if (item_data == NULL) {
+		ShowError("buildin_delitemidx: Invalid Item ID (%d).\n", it.nameid);
+		st->state = END;
+		return false;
+	}
+
+	it.identify = sd->status.inventory[i].identify;
+	it.refine = sd->status.inventory[i].refine;
+	it.attribute = sd->status.inventory[i].attribute;
+	it.card[0] = sd->status.inventory[i].card[0];
+	it.card[1] = sd->status.inventory[i].card[1];
+	it.card[2] = sd->status.inventory[i].card[2];
+	it.card[3] = sd->status.inventory[i].card[3];
+	
+	if (script_hasdata(st, 3)) {
+		if ((it.amount = script_getnum(st, 3)) > sd->status.inventory[i].amount)
+			it.amount = sd->status.inventory[i].amount;
+	} else {
+		it.amount = sd->status.inventory[i].amount;
+	}
+
+	// nothing to do
+	if (it.amount <= 0)
+		return true;
+
+	// success
+	if (!script->buildin_delitem_search(sd, &it, true)) {
```

Comment:
If you have already inventory index and this mean you have access to all item fields/counts etc and can change it directly. If want reuse existing delete code, can be used script->buildin_delitem_delete

Reply:
good point ! why I miss that XD